x-pack/plugins/ml/public/application/jobs/jobs_list/components/datafeed_chart_flyout/fill_missing_chart_data.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * Returns an array with the filled in missing timestamp values for scatterplot charts.
+ * @param dataWithPossibleMissingValues - An array of arrays consisting of [timestamp, doc_count]
+ * @param dataWithAllValues - An array of arrays consisting of [timestamp, doc_count]
+ */
+
+export type ChartDataWithNullValues = Array<[number, number | null]>;
```

Comment:
You could move `[number, number | null]` to its own type definition like `type ChartDataItem = [number, number | null]` so it can be reused further down in this file.